File: frontend/src/types/conversation.ts
export type MessageType = 'text' | 'image' | 'file' | 'system_event'
export type ConversationStatus = 'open' | 'closed'

export interface Conversation {
  id: string
  status: ConversationStatus
  startedAt: string       // ISO 8601
  endedAt: string | null
  agentName: string
  customerName: string
  messageCount: number
}

export interface Message {
  id: string
  conversationId: string
  type: MessageType
  content: string
  senderRole: 'customer' | 'agent' | 'system'
  sentAt: string          // ISO 8601
}

File: backend/app/models/conversation.py
from typing import Literal
from pydantic import BaseModel


class Conversation(BaseModel):
    id: str
    status: Literal["open", "closed"]
    startedAt: str
    endedAt: str | None
    agentName: str
    customerName: str
    messageCount: int


class Message(BaseModel):
    id: str
    conversationId: str
    type: Literal["text", "image", "file", "system_event"]
    content: str
    senderRole: Literal["customer", "agent", "system"]
    sentAt: str


class PaginatedConversations(BaseModel):
    data: list[Conversation]
    total: int
    page: int
    page_size: int


class PaginatedMessages(BaseModel):
    data: list[Message]
    total: int
    page: int
    page_size: int
